sequential/bank_accounts/src/bank_account.erl

```diff
@@ -2,5 +2,16 @@
 
 -export([process_operation/2]).
 
-process_operation(Bank, Operation) ->
-  put_your_solution_here.
+process_operation(Bank,
+		  {AccNumber, withdraw, Amount}) ->
+    case lists:keyfind(AccNumber, 1, Bank) of
+      false -> {error, account_not_found};
+      {AccNumber, Funds} when Funds < Amount ->
+	  {error, insufficient_funds};
+      {AccNumber, Funds} -> {AccNumber, Funds - Amount}
+    end;
+process_operation(Bank, {AccNumber, deposit, Amount}) ->
+    case lists:keyfind(AccNumber, 1, Bank) of
```

Comment:
you could do account finding (which is the same in both cases) in a helper function

Reply:
Refactored 78aa5cce96a9d7c2bfdda7c93860d8119477a4ea